src/stdlib.c

```diff
@@ -239,7 +239,27 @@ Array.prototype.indexOf = function(a, b) {
     }
   }
   return -1;
-}
+};
+
+Array.prototype.lastIndexOf = function(a, b) {
+  if (!b || b < 0 || b >= this.length) b = this.length - 1;
+  for (var i = b; i >= 0; i--) {
+    if (this[i] == a) {
+      return i;
+    }
+  }
+  return -1;
+};
+
+Array.prototype.xmap = function(a, b) {
```

Comment:
Dead code removed.
JS implementation is wrong. If JS array has gaps (e.g. a = [1]; a[100] = 2;), then "for" loop that iterates over includes all indices, whereas only existing ones should be used.
ptal